synapseclient/core/utils.py

```diff
@@ -339,15 +339,46 @@ def is_same_base_url(url1: str, url2: str) -> bool:
     return url1.scheme == url2.scheme and url1.hostname == url2.hostname
 
 
-def is_synapse_id_str(obj):
+def is_synapse_id_str(obj: str) -> typing.Union[str, None]:
     """If the input is a Synapse ID return it, otherwise return None"""
     if isinstance(obj, str):
-        m = re.match(r"(syn\d+$)", obj)
+        m = re.match(r"(syn\d+(\.\d+)?$)", obj)
         if m:
             return m.group(1)
     return None
 
 
+def get_synid_and_version(
+    obj: typing.Union[str, collections.abc.Mapping]
+) -> typing.Tuple[str, typing.Union[int, None]]:
+    """Extract the Synapse ID and version number from input entity
+
+    Returns:
+        A tuple containing the synapse ID and version number,
+        where the version number may be an integer or None if
+        the input object does not contain a versonNumber or
+        .version notation (if string).
+    """
+
+    if isinstance(obj, str):
+        synapse_id_and_version = is_synapse_id_str(obj)
+        assert (
```

Comment:
In a few other functions in this utils class it is doing a `raise ValueError(msg)` rather than an assert. What's your opinion on one or the other?

Reply:
For example the closest in functionality is `id_of` which does a:
`raise ValueError("Invalid parameters: couldn't find id of " + str(obj))`

Reply:
Thanks for pointing this out - I think keeping with the `ValueError(msg) is better for consistency, and the error type is more specific. I'll make a commit to replace this `assert`.